src/contexts/useArchaeologistService.ts

```diff
@@ -0,0 +1,93 @@
+      p: '8xWEC27ReEeXfsy_avEZEtZPisuogXU9EElM6rXvYnCt91gwH-duVv4RNWLA_oiWuhr0cFLy8CUmnt-RzVQvswJglB8FAadm1-lS4oohX0qcnu50xK6zhrPU02q9pIepVsU2uhBjsLox_HORhvQcY_GEZndWip7cTZ0OUNiePIfK5DidaHG6c4bA_KHxgBNwiH5wt1p9E1F1wXNd7qKFhIUZXhbMb5i0nIH45-pMCGY9xMChx2cKvLEpOTdwG-oDh6a6OpQ65iQJTjzcZQtSjGaARdZ2vP6e18Mtk3kN_mKEWwoR1f0pDNcAdrY6ZtWOPkK63JAOxFOY3M3MCjUqyw',
+      q: '8ZCttPunWAiPcWqWFLgCh22wuJRJzz_qBPo_-6tB58dOuIIpyRmUNisjwGFvHN-g6Uv_tLMzpQ7_oMetwYWpBaENpfbcnzAIssoT4qB3lTNqvWjd3OZFJJOfZObfr8J12Hb2eDcEI_9ZVk_84Otq-gI6_5mkziFex8zyQ5FlOW6FaE7Gvxv8HBcBs9QeAmMuulQVD25ovhWkILyQKMAFaAx8vxH9HsElnPuthV6xLV2aI11nrYJDUwUCRK-lfdCos0y5r_hOzrXvIW6PZ5FqnSPJAwIC_sWw8FXWvfsVVdi6FoOId6kHpCmNFrVVZyloBBjZ2Tvq4Ws-DywUKEBgBw',
+      dp: '2TWTnSez81d9jqdkMCop7oQj5XEgLDXXj7MAEQbrVjo7qTZbIlcJaKB0PhD1RTxF-xxU6f2k6WM5BGPgA-rDCKEKLbEeJyEvq6TpFFa7hUvFcdq_6nP1JVuRxCywMtGqRkfJJC-69NMcEWovUrjUmZXZajoe0ZUsZAoAHcvCcXIhoRbZyNc-9TYnQldu1bQJLV5WV_B6rQVwRKFTAQamagkI5uNWYC3Czcrx4EGjGSpK7YH3zZdMp9libacU_Ux5RQB9jhSgIL94tTl68qtQYvbAjicaSLz4sPHNyuYz0-iyPitcTqt9v-RqnKkuvG79QLMeCDKNj27nu_4E4eYqSw',
+      dq: 'wOg4z_7ndHXhl0D2-TS5UE1Ygw5oowj3aWIvFlt9Fgsj4lSP6fhzU46_zjIA-Z3ccnIuka9J3tZLbaoIKDME92nV7GNna_h_-cdGegSLeqrE4t2Lz4xUSxFYiMuWdlVkpqjZysakl9qjDTpzR2q4005UrymTYsljV0tjbHH_1A1enB3MuXEvs37r7R_Hl0B9YG8lIl2ZZ7MAjuQGe4RczOqr9Cn5aUX1M5gIyikEdKX7IaLF2Nndyb6GQTPDhOXM0uLCprQ0uv6vKJOSEbMe4L5jF_PTUIGu-z6RbsXEE_ovISuJH4XshDxqX2gP-4IlfqAZ2FDxEXMC7DrXHg3KoQ',
+      qi: 'kS2AXYKfVKOdtNGiksIgs2YT4Pz-62rCeynyFlnUDc6E3eylzctLHhuWMRqWqnPwVPwnw2Fd6eUTOYZe82kVf9d0-XbeDtOOP6fS7HU30ICIvFrLTiaVEsLekUdQwoyC7s0xH99FZpdGJHzC7iNdlChybmrh4Ci5OyOLn_o8goNQBYWEXOgr6JgJLTh9BXvDEbNLLJ0JTZFQeLVqQeimPDSoAwBFCLBBwOd6CFOr5wTc9Pkt8cDrpcf9usU-61uu3Lr4-3PG4yUUQyphnDy84DgpUyuF5JNy0KgtuZaMMuAZJTb_gKN3IdLUHV9phqwrSKylLn-enGJ0f4FTpfcDEQ',
+    };
+
+    const address = await arweave.wallets.getAddress(key);
+    console.log(
+      'start upload for sarcophagus:',
+      address,
+      sarcoId,
+      await arweave.wallets.getBalance(address)
+    );
+    const tx = await arweave.createTransaction({ data: file }, key);
+    tx.addTag('Content-Type', 'plain/text');
+    await arweave.transactions.sign(tx, key);
+
+    let uploader = await arweave.transactions.getUploader(tx);
+
+    while (!uploader.isComplete) {
+      await uploader.uploadChunk();
+      console.log(
+        `${uploader.pctComplete}% complete, ${uploader.uploadedChunks}/${uploader.totalChunks}`
+      );
+    }
+    return tx.id;
+  };
+
+  const updateStatus = async () => {
+    const arweave = initArweave();
+    const currentTextArweaveAddress = 'Xm17-cZJjcx-jc_UL5me1o5nfqC2T1mF-yu03gmKeK4';
+    const txId = await arweave.wallets.getLastTransactionID(currentTextArweaveAddress);
+    console.log('txId:' + txId);
+
+    const res = await arweave.transactions.getStatus(txId);
+    if (res && res.status === 200) {
+      if ((res.confirmed as unknown as string) === 'Pending') {
```

Comment:
Yeah that one. I wrote that based on the TS comments, but I honestly don't understand the typings for the confirmed property. In once case it returns a string, and in another it returns TransactionConfirmedData type. but it's only typed to the later. I think I could find a better test for when confirmed is set ....

Reply:
@mobiusTripper-crypto -- would recommend checking for presence of `res.confirmed` in the `if (res && res.status === 200)` conditional. That should make the typescript error go away. Typescript is saying it could be null at that point, so we do want to check that it isn't.